core/embed/trezorhal/unix/touch.c

```diff
@@ -53,227 +51,230 @@ typedef enum {
   BUTTON_SWIPE_UP_INITIATED,
   BUTTON_SWIPE_DOWN_INITIATED,
   BUTTON_SWIPE_COMPLETED
-} touch_input_state_t;
+} touch_state_t;
+
+typedef struct {
+  // Set if driver is initialized
+  secbool initialized;
+  // Current state of the touch driver
+  touch_state_t state;
+  // Last valid coordinates
+  int last_x;
+  int last_y;
 
-static touch_input_state_t input_state = IDLE;
+} touch_driver_t;
+
+// Touch driver instance
+touch_driver_t g_touch_driver;
 
 static bool is_inside_display(int x, int y) {
   return x >= sdl_touch_offset_x && y >= sdl_touch_offset_y &&
          x - sdl_touch_offset_x < sdl_display_res_x &&
          y - sdl_touch_offset_y < sdl_display_res_y;
 }
 
-static bool is_button_swipe_initiated() {
-  return input_state == BUTTON_SWIPE_LEFT_INITIATED ||
-         input_state == BUTTON_SWIPE_RIGHT_INITIATED ||
-         input_state == BUTTON_SWIPE_UP_INITIATED ||
-         input_state == BUTTON_SWIPE_DOWN_INITIATED;
+static bool is_button_swipe_initiated(touch_driver_t* driver) {
```

Comment:
driver can be pointer to const

Reply:
yes, fixed, see 38441e0bc9e4a44a5a2cafff09a1118ca5142d7d